src/ScrollingContainer.js

```diff
@@ -270,15 +266,44 @@ ScrollingContainer.prototype.initScrolling = function () {
         container.position[direction] = Math.round(Position[direction]);
 
         self.updateScrollBars();
-
     };
 
-    this._checkLimit = function (limitName) {
-        if (!(limitName in limitsReached)) {
-            limitsReached[limitName] = true;
-            self.emit(limitName + "Overflow");
+    this._checkLimits = function (direction) {
+        if (!this.scrolling && Math.round(Speed[direction]) === 0) {
+            return;
         }
-    };
+
+        var bounds = this.getInnerBounds();
+
+        var minimumLimit = (direction == "y") ?
+            Math.round(Math.min(0, this._height - bounds.height)) :
+            Math.round(Math.min(0, this._width - bounds.width));
+
+        var maxLimit = targetPosition[direction] >= 0;
+        var minLimit = targetPosition[direction] <= minimumLimit;
+
+        if (direction === "x" && maxLimit) {
+            self.emit("leftReached");
+        } else if (direction === "x" && minLimit) {
+            self.emit("rightReached");
+        } else if (direction === "y" && maxLimit) {
+            self.emit("topReached");
+        } else if (direction === "y" && minLimit) {
+            self.emit("bottomReached");
```

Comment:
@Nahikariii I've changed the variable names to actually match what they are testing and renamed the events to something less ambiguous.